### PR TITLE
add functionality on Home and Products page

### PR DESCRIPTION
**Pull request by hsnice16**
### This PR will add code for the functionality of the filters on the product page and links the products page from home.

Features this PR will add:
1. On click of the Category card on the Home page, it navigates to the Product page and shows the products based on the selected category
1. You can navigate to the Products page from the footer link also
1. You can navigate to the Home page, either clicking on the logo or using the footer link
1. On the Products page, you can filter data based on the filter options
1. On selecting any filter, the selected filter is also reflected in the URL of the page. And you can share that URL with anyone, to show your filtered preference
1. Code is not in this PR, but you can also check that on each refresh the Products in the Shop Now section on the Home page contains different Products. 

In this PR, **615 additions 85 deletions** lines of code
- many files contain only 5 to 6 lines of the code
- if you don't prefer, then you can ignore to review CSS files and Backend related files

#### [Live Site](https://react-inclusive-mart-git-feat-functionality-hsnice16.vercel.app/)

### CODE REVIEW

**Review by zoef-yt — APPROVED** [review]
The code is really code and well written. Nice use of comments everywhere. Good to merge. Great work Himanshu.

File: src/components/products-filter-list/ClearAllButton.jsx [hunk 1]
```diff
@@ -1,21 +1,24 @@
 import PropTypes from "prop-types";
```
Comment (mikesch803) [inline src/components/products-filter-list/ClearAllButton.jsx:1]: This is nice, you are using prop-types

File: src/custom-hooks/useFilteredData.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { useProducts } from "../context";
+import {
+  getMinimumSelectedRating,
+  getNumberFromFormattedPrice,
+  getSortingFunction,
+  removeEmptyStringsFromArray,
+} from "../utils";
```
Comment (mikesch803) [inline src/custom-hooks/useFilteredData.js:7]: you separate out each function, this is good

File: src/utils/functions/getFilterStateFromSearchParams.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+/**
+ * getFilterStateFromSearchParams - function to get the
+ *                               filter initial state
+ *
+ * @param {Array} searchParams - value of the all
+ *                              search query strings
+ * @returns state object
+ */
```
Comment (mikesch803) [inline src/utils/functions/getFilterStateFromSearchParams.js:36]: you specified the multi comment that is nice, a viewer can easily understand this

File: src/pages/products/shared/ProductsFilterList.jsx [hunk 4]
```diff
@@ -1,36 +1,40 @@
...
 
-  return [status, productStatus].includes("loading") ? (
+  const {
+    handleCategoryRadioChange,
+    handleSubCategoryCheckBoxChange,
+    handleRatingsCheckBoxChange,
+    handleSortByPriceRadioChange,
+    handlePriceRangeChange,
+    handleResetBtnClick,
+  } = useFilterHandler();
```
Comment (mikesch803) [inline src/pages/products/shared/ProductsFilterList.jsx:35]: good you are using custom hooks, separate out the functions

**Review by mikesch803 — APPROVED** [review]
you wrote code very well, you can merge this

**Comment by hsnice16** [comment]
Thank you for reviewing.

**Pull request merged**